### PR TITLE
edit: Add automatic upgrade policy

### PR DESCRIPTION
**Pull request by vkareh**

### CODE REVIEW

**Comment by wgordon17** [comment]
Hey @vkareh, I have a general _feeling_ that this is a better fit for `rosa edit cluster`, since this is about setting a _policy_, as opposed to actually actioning an _upgrade_.

**Comment by vkareh** [comment]
that's not a bad idea. I'll update this

**Comment by jharrington22** [comment]
/hold

@vkareh adding this in light of the STS upgrade discussions.

**Comment by vkareh** [comment]
@jharrington22 I've updated this to block setting automatic upgrade schedules to STS clusters.

**Comment by vkareh** [comment]
@wgordon17 @jharrington22 - now that OCM only does automatic upgrades for z-streams this can probably be unblocked and merged. What do you think?

**Comment by tzvatot** [comment]
@vkareh this has been stale for a while. Do you like to revive it?

**Comment by tzvatot** [comment]
@vkareh MR has conflicts. Can you fix those?

File: cmd/edit/cluster/cmd.go [hunk 1]
```diff
@@ -48,14 +51,41 @@ var args struct {
 	httpsProxy                string
 	noProxySlice              []string
 	additionalTrustBundleFile string
+
+	// Upgrade schedule options
+	upgradeScheduleDay  string
+	upgradeScheduleTime string
+}
+
+var dayOptions = []string{
```
Comment (tzvatot) [inline cmd/edit/cluster/cmd.go:60]: Do we really need this an also `daysMap`? seems like a duplication. We can get the map keys (keys := reflect.ValueOf(daysMap).MapKeys())

**Pull request closed without merging**